compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ProjectFiles.java

```diff
@@ -230,6 +233,9 @@ public static void validateSingleFileProjectFilePath(Path filePath) {
         }
 
         if (!Files.isRegularFile(filePath) || !ProjectFiles.BAL_EXTENSION_MATCHER.matches(filePath)) {
+            if (ProjectFiles.JAR_EXTENSION_MATCHER.matches(filePath)) {
```

Comment:
This will give the wrong error message in other commands (e.g. build, test, doc, etc) if the user passes a jar to those. I think it would be better to have it in the run command.

Reply:
I moved the check exclusively to the RunCommand class. Is the implementation correct?

Reply:
yes, looks good